### ISSUE TITLE
[Feature Request] Disable Unused Factories

### ISSUE THREAD

**Issue by batty1991** [tobspr/shapez.io#1043]
Implement a function that allows the user to disable and not delete certain factories that are not currently necessary. The idea is to free up processing and increase FPS by removing large factories from the current processing chain.

**Comment by darkwater4213** [comment]
You *could* just disconnect them. Or am I misunderstanding your problem? You can also increase the tick rate, if that helps.

**Comment by batty1991** [comment]
My intention was to be able to ‘disable’ the factories in an attempt to reduce the processing/rendering requirements of idle factories.
I have quite a few large factories that I rotate around depending on the pieces I need. I’m using: 161k belts, 256k buildings.
I get between 3 & 5 FPS. I have been refining my techniques with no meaningful results.

My rig is no slouch: AMD Threadripper 2920X, 32Gb DDR4 @ 3200MHZ, RTX 2080 Super

Josh Hoffman

**Comment by darkwater4213** [comment]
Huh. I think I see where you’re going with this... Basically, have a separate “archive” save file that only loads a single chunk at a time onto the default save, and doesn’t show up in the saves menu. All edits to the default save would automatically be copied over to the archived version, and you can also click a button on the HUD labeled “archive buildings” that removes them from the default but not the archive. Am I getting this somewhat right?

**Comment by batty1991** [comment]
I’m sure about the logistics you are suggesting. In different terms, I imagine an additional Boolean attribute for each building and belt such as: ‘active’. The user would have the option to change this attribute in the HUD (in mass). The game would use the attribute when evaluating an object and skip ‘active’=false objects in the normal processing/rendering threads. Thus alleviating resource load and improving game performance.

Josh Hoffman

**Comment by darkwater4213** [comment]
> I’m sure about the logistics you are suggesting. In different terms, I imagine an additional Boolean attribute for each building and belt such as: ‘active’. The user would have the option to change this attribute in the HUD (in mass). The game would use the attribute when evaluating an object and skip ‘active’=false objects in the normal processing/rendering threads. Thus alleviating resource load and improving game performance. 

Yeah, that's more or less what I was saying. I tend to needlessly overcomplicate things, but that does sound like a good idea. I think it might be a little to much for @tobspr to handle right now, but it does remind me of a different but related idea: Each savegame, instead of being a single `.bin` file, would be a `.zip` archive containing three files: `archived.bin`, `<name>.bin`, and `settings.json` (or similar) and maybe a folder labeled `blueprints`, containing the binaries for, say, the ten most-used blueprints for that savegame. This would (while probably taking a lot of time to code) would resolve multiple feature requests. (I'll edit the links into here as soon as I find them.)

**Comment by immibis** [comment]
This is a silly design for the feature. What you really want is that the game automatically sets things to inactive when they aren't doing anything, and then doesn't waste time processing them.
